src/Akka.Hosting/Logging/LoggerSetup.cs

```diff
@@ -0,0 +1,108 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="LoggerSetup.cs" company="Akka.NET Project">
+//      Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Event;
+
+namespace Akka.Hosting.Logging
+{
+    public class LoggerSetup
+    {
+        private readonly List<Type> _loggers = new List<Type> { typeof(DefaultLogger) };
+        internal AkkaConfigurationBuilder Builder { get; }
+
+        internal LoggerSetup(AkkaConfigurationBuilder builder)
+        {
+            Builder = builder;
+        }
+
+        /// <summary>
+        /// <para>
+        /// Log level used by the configured loggers.
+        /// </para>
+        /// Defaults to <c>LogLevel.InfoLevel</c>
+        /// </summary>
+        public LogLevel LogLevel { get; set; } = LogLevel.InfoLevel;
+        
+        /// <summary>
+        /// <para>
+        /// Log the complete configuration at INFO level when the actor system is started.
+        /// This is useful when you are uncertain of what configuration is being used by the ActorSystem.
+        /// </para>
+        /// Defaults to false.
+        /// </summary>
+        public bool LogConfigOnStart { get; set; } = false;
+
+        /// <summary>
+        /// Clear all loggers currently registered.
+        /// </summary>
+        /// <returns>This <see cref="LoggerSetup"/> instance</returns>
+        public LoggerSetup ClearLoggers()
+        {
+            _loggers.Clear();
+            return this;
+        }
+
+        /// <summary>
+        /// Register a logger
+        /// </summary>
+        /// <returns>This <see cref="LoggerSetup"/> instance</returns>
+        public LoggerSetup AddLogger<T>() where T: ActorBase
+        {
+            var logger = typeof(T);
+            var methodInfo = logger.GetMethod(
```

Comment:
This reflection is used to get the static "magic method" called `ConfigureHosting` inside the logger class, if we found this static method, we will call it so the logger can register itself properly with Akka.Hosting

Reply:
I think I still need to fix this - it's not covered by unit tests at the moment.

Reply:
Actually, looks like we don't need it.